### DISCUSSION TITLE
懒加载元素未渲染时引起的数据和校验问题

### DISCUSSION THREAD

**Discussion by daijingfeng** [alibaba/formily#3841] [Q&A 2.x]
### Demo复现
[codesandbox](https://codesandbox.io/s/compassionate-ben-qiygu1?file=/src/index.tsx)
### 场景
表单是一个双层Tab嵌套的形态，默认只展开第一个Tab项。组件是自定义实现。demo是简化的版本，实际每个表单项复杂很多，如果一次性渲染完，性能会有影响，因此需要做懒加载。但实现懒加载后带来了两个问题。
### 问题1：数据问题
当使用 `InterSectionObserve` 实现元素懒加载后，未渲染的元素，通过 `form.values` 获取的数据不完整。

例如该表单是一个默认 2个 Tab，每个 Tab 8 条数据。**全部渲染完成时**，获取的数据是正确的。
```json
{"items":[
  {
    "items":[
      {"input":""},{"input":""},{"input":""},{"input":""},{"input":""},{"input":""},{"input":""},{"input":""}
    ]
 },
 {
   "items":[
    {"input":""},{"input":""},{"input":""},{"input":""},{"input":""},{"input":""},{"input":""},{"input":""}
   ]
 }
]}
```
**当只渲染了两个元素时**，只获得是两个元素的数据
```json
{"items": [
   {
     "items":  [
         {"input":""},{"input":""}
     ]
  },
  {
     "items":[]
  }
]}
```
### 问题2：校验问题
通过 `form.validate` 触发校验也仅是对已经渲染的元素的校验。想要实现的是通过校验方法能对**整个表单进行校验，即使元素未渲染**。并且元素在**可视区域显示**或**Tab展开**时，能正常显示错误信息。

🙏麻烦大佬帮忙看看这种场景如何解决。

**Comment by beita-wang** [comment]
请教一下这个问题解决了么 我遇到了同样的问题，组件都是懒加载的，form.query字段的时候 有些组件还没加载，导致没办法做初始化。

**Reply by daijingfeng** [reply]
通过 form.setInitialValues 设置初始值达到了我的需求
